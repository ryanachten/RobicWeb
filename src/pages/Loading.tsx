import * as React from "react";
import { Theme } from "@material-ui/core/styles/createMuiTheme";
import createStyles from "@material-ui/core/styles/createStyles";
import withStyles, { WithStyles } from "@material-ui/core/styles/withStyles";
import { Classes } from "jss";
import { CircularProgress, Typography } from "@material-ui/core";

const styles = (theme: Theme) =>
  createStyles({
    root: {
      alignItems: "center",
      display: "flex",
      flexDirection: "column",
      justifyContent: "center",
      height: "100vh",
      width: "100vw"
    },
    title: {
      marginBottom: theme.spacing.unit * 4
    }
  });

type State = {};

type Props = {
  classes: Classes;
};

class Loading extends React.Component<WithStyles<typeof styles>, State> {
  constructor(props: Props) {
    super(props);
  }

  render() {
    const classes = this.props.classes;
    return (
      <div className={classes.root}>
        <Typography className={classes.title} variant="h2" component="h1">
          hold tight, we're loading...
        </Typography>
        <CircularProgress />
      </div>
    );
  }
}

export default withStyles(styles)(Loading);
