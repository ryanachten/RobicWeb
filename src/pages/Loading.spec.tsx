import React from "react";
import { createShallow } from "@material-ui/core/test-utils";
import Loading from "./Loading";
import renderer from "react-test-renderer";

describe("Page: Loading", () => {
  let shallow: any;

  beforeEach(() => {
    shallow = createShallow({ untilSelector: "Loading" });
  });

  it("renders correctly", () => {
    const tree = renderer.create(<Loading />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
