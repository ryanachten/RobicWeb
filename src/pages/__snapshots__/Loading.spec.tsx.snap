// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: Loading renders correctly 1`] = `
<div
  className="Loading-root-1"
>
  <h1
    className="MuiTypography-root-3 MuiTypography-h2-16 Loading-title-2"
  >
    hold tight, we're loading...
  </h1>
  <div
    className="MuiCircularProgress-root-39 MuiCircularProgress-colorPrimary-42 MuiCircularProgress-indeterminate-41"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg-44"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle-45 MuiCircularProgress-circleIndeterminate-47"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;
